src/Disks/IO/createReadBufferFromFileBase.cpp

```diff
@@ -129,11 +129,7 @@ std::unique_ptr<ReadBufferFromFileBase> createReadBufferFromFileBase(
 
         auto align_up = [=](size_t value) { return (value + min_alignment - 1) / min_alignment * min_alignment; };
 
-        if (alignment == 0)
-            alignment = min_alignment;
-        else if (alignment % min_alignment)
-            alignment = align_up(alignment);
-
+        size_t buffer_alignment = alignment == 0 ? min_alignment : align_up(alignment);
```

Comment:
Previously we could align alignment and not align the size of buffer and in case of failure of creation of buffer with `O_DIRECT` in fallback case the logical error `Too large alignment` could be thrown.